cmd/publishing-bot/config/rules.go

```diff
@@ -0,0 +1,64 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Dependency of a piece of code
+type Dependency struct {
+	Repository string `yaml:"repository"`
+	Branch     string `yaml:"branch"`
+	// Dir from repo root
+	Dir string `yaml:"dir,omitempty"`
+}
+
+func (c Dependency) String() string {
+	repo := c.Repository
+	if len(repo) == 0 {
+		repo = "<source>"
+	}
+	return fmt.Sprintf("[repository %s, branch %s, subdir %s]", repo, c.Branch, c.Dir)
+}
+
+type BranchRule struct {
+	Name string `yaml:"name"`
+	// k8s.io/* repos the branch rule depends on
+	Dependencies []Dependency `yaml:"dependencies,omitempty"`
```

Comment:
follow-up: split Dependency and Source. The `Dir` doesn't make sense for the former.

Reply:
https://github.com/kubernetes/publishing-bot/pull/25